tests/test_user_error_output.py

```diff
@@ -96,3 +108,19 @@ def test_failing_optional(capture_output, args):
     assert (
         f"Incorrect value for {args[0]} (-10). Input should be greater than 0"
     ) in capture_output.captured_output
+
+
+class OptionalValueOldTyping(BaseModel):
```

Comment:
The issue you found revolves about cli optional arguments. 

I would say the best place to put any test would be in `test_optionals.py`.

something like this:

```python

class Model(BaseModel):
    value: int | None = None


def test_optional_with_none():
    model = parse(Model, ["--value", "10"])
    assert model == Model(value=10)


class OldModel(BaseModel):
    value: Optional[int] = None


def test_old_model_with_none():
    model = parse(OldModel, ["--value", "10"])
    assert model == OldModel(value=10)
```

Reply:
done!